cloudify_gcp/compute/tests/test_instance.py

```diff
@@ -199,6 +199,55 @@ def test_create_2_boot_disks_raises(self, *args):
         for name in 'disk_1', 'disk_2':
             self.assertIn(name, str(e.exception))
 
+    def test_create_with_metadata(self, mock_build, *args):
+        metadata = {'metadata': {'items': [{'key': 'k', 'value': 'v'}]}}
+        instance.create(
+            'type',
+            'image',
+            'name',
+            external_ip=True,
+            startup_script=None,
+            scopes='scopes',
+            tags=['tags'],
+            additional_settings=metadata
+        )
+
+        mock_build().instances().insert.call_args[1][
+            'body']['tags']['items'].sort()
+        mock_build().instances().insert.assert_called_with(
+            body={
+                'machineType': 'zones/a very fake zone/machineTypes/type',
+                'tags': {'items': ['name', 'tags']},
+                'canIpForward': False,
+                'metadata': {'items': [
+                    {'value': 'not really a project',
+                     'key': 'bucket'},
+                    {'value': '',
+                     'key': 'sshKeys'},
+                    {'key': 'k',
+                     'value': 'v'}
```

Comment:
These need to be in the same order that the code adds them, as the comparison in the assert is strict